_playwright-tests/tests/API/apiHelpers.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  BulkDeleteRepositoriesRequest,
+  Configuration,
+  DeleteTemplateRequest,
+  ListTemplatesRequest,
+  RepositoriesApi,
+  TemplatesApi,
+  type ListRepositoriesRequest,
+} from './client';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// while condition is true, calls fn, waits interval (ms) between calls.
+// condition's parameter should be the result of the function call.
+export const poll = async (
+  fn: () => Promise<any>,
+  condition: (result: any) => boolean,
+  interval: number,
+) => {
+  let result = await fn();
+  while (condition(result)) {
+    result = await fn();
+    await timer(interval);
+  }
+  return result;
+};
+
+// timer in ms, must await e.g. await timer(num)
+export const timer = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+export const cleanupRepositories = async (
```

Comment:
if things fail during cleanup, it might be difficult to understand where or why with approach 1 (without the test.step). what do you think about adding a try/catch in the cleanup functions and/or some logging?

Reply:
the cleanup functions should never fail / throw exceptions as they are I think 🤔 but if they do, then explicitly ignoring all errors with try/catch could be a helpful 😅 
_and I improved them a bit since you commented so have a look 😇_